src/services/expenses.service.js

```diff
@@ -0,0 +1,63 @@
+const expenses = [];
+const generateId = (arr) => arr.length + 1;
+
+const getAll = () => expenses;
+
+const resetExpenses = () => (expenses.length = 0);
+
+const remove = (index) => expenses.splice(index, 1);
+
+const indexOfExpense = (id) =>
+  expenses.findIndex((expense) => expense.id === +id);
+
+const update = (index, data) => {
+  const expense = expenses[index];
+
+  const updatedFields = Object.keys(data).reduce((acc, key) => {
+    acc[key] = data[key];
+
+    return acc;
+  }, {});
+
+  expenses[index] = { ...expense, ...updatedFields };
+
+  return expenses[index];
+};
+
+const create = (data) => {
+  const newExpense = { ...data, id: generateId(expenses) };
+
+  expenses.push(newExpense);
+
+  return newExpense;
+};
+
+const filterByQuery = (category, from, to, userId) => {
+  const fromDate = from ? new Date(from) : null;
+  const toDate = to ? new Date(to) : null;
+
+  return expenses.filter((expense) => {
+    const mathesCategory = category ? expense.category === category : true;
```

Comment:
There is a typo in the variable name `mathesCategory`. It should be `matchesCategory` to maintain consistency and clarity.